src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -235,14 +241,14 @@ ReplaceTableVisibleFunctionWalker(Node *inputNode)
 			 */
 			functionToProcess->funcid = CitusTableVisibleFuncId();
 
-			return (Node *) functionToProcess;
+			return false;
```

Comment:
Minor: Technically speaking, you could have nested `pg_table_is_visible` function calls. Should we just drop into the expression_tree_walker here?

Reply:
Probably not a very likely case but added the call and a test as well.